### PR TITLE
Add index-based filtering to table listing

### PR DESCRIPTION
**Pull request by amorenoz**
For now, only support index based filtering, i.e: [INDEX]=[VALUE]. More complex filtering would require https://github.com/ovn-org/libovsdb/pull/209 to get merged

Fixes: #25

### CODE REVIEW

**Comment by lewisdenny** [comment]
> @lewisdenny can you please try this PR? You should be able to use the image as per the instructions in the Readme.md using this image: quay.io/amorenoz/ovsdb-mon:filters

Hey Adrián,
Could you please provide an example command to run? I can't seem to get it to work and don't see it in the help menu either.
```
>>> list Logical_Switch_Port help

List the content of Table Logical_Switch_Port

Logical_Switch_Port [Field1 Field2 ...]
        [Field1 Field2 ...]: List of fields to show (default: all fields will be shown)
                Possible Fields: UUID, Addresses, Dhcpv4Options, Dhcpv6Options, DynamicAddresses, Enabled, ExternalIDs, HaChassisGroup, Name, Options, ParentName, PortSecurity, Tag, TagRequest, Type, Up
```
```
[root@overcloud-controller-0 ~]# podman images | grep ovs
quay.io/amorenoz/ovsdb-mon                                                    filters      3e03a66b5759   12 hours ago   1.26 GB
```

**Comment by amorenoz** [comment]
You're right @lewisdenny , sorry. The container is not picking up the code it should. Let me fix that real quick

**Comment by amorenoz** [comment]
Rebased this branch on top of #31 which fixes the image generation.
@lewisdenny you can now try the latest "fixes" tag

File: cmd/ovsdb-mon/shell.go [hunk 1]
```diff
@@ -338,3 +373,31 @@ func colordiff(a, b interface{}) string {
...
+	}
+	parts := strings.Split(expr, "=")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("Invalid filter expression: %s", expr)
+	}
+
+	field := parts[0]
+	value := parts[1]
+
+	fieldVal := reflect.ValueOf(condModel).Elem().FieldByName(field)
```
Comment (flavio-fernandes) [inline cmd/ovsdb-mon/shell.go:337]: heh... can you please make this case insensitive? I'm getting spoiled. ;)

File: cmd/ovsdb-mon/shell.go [hunk 2]
```diff
@@ -234,18 +234,40 @@ func (s *OvsdbShell) Run(ovsPtr *client.Client, args ...string) {
 			Help:    fmt.Sprintf("%s [Field1 Field2 ...]", name),
 			LongHelp: fmt.Sprintf(
 				"List the content of Table %s", name) +
-				fmt.Sprintf("\n\n%s [Field1 Field2 ...]", name) +
+				fmt.Sprintf("\n\n%s [--filter Field=Value] [Field1 Field2 ...]", name) +
```
Comment (flavio-fernandes) [inline cmd/ovsdb-mon/shell.go:236]: nice! I think we are also missing a similar update in the output of the --help command:
For example:
```
ovsdb-mon list --help
2021/11/07 23:05:41 libovsdb: trying to connect to DB unix:///run/openvswitch/ovnnb_db.sock
2021/11/07 23:05:41 libovsdb: connected to unix:/run/openvswitch/ovnnb_db.sock
OVSDB Monitoring Shell

List the content of a specific table
Commands:
  ACL                              ACL [Field1 Field2 ...]   <--- pls add --filter text here!
```

File: cmd/ovsdb-mon/shell.go [hunk 3]
```diff
@@ -338,3 +318,148 @@ func colordiff(a, b interface{}) string {
...
+					c.Println("Only one --filter statement allowed")
+					return
+				}
+				filter = arg
+				isFilter = false
+			} else {
+				if col, exists := s.exactFieldName(tableName, arg); exists {
+					columns = append(columns, col)
+				} else {
+					c.Println("Field %s not found in table %s", arg, tableName)
```
Comment (flavio-fernandes) [inline cmd/ovsdb-mon/shell.go:408]: c.Println("Field", arg, "not found in table", tableName)

of maybe use Printf?

```
ovsdb-mon.nb list Logical_Switch_Port --help
2021/11/07 23:15:53 libovsdb: trying to connect to DB unix:///run/openvswitch/ovnnb_db.sock
2021/11/07 23:15:53 libovsdb: connected to unix:/run/openvswitch/ovnnb_db.sock
OVSDB Monitoring Shell
Field %s not found in table %s --help Logical_Switch_Port
```

**Review by flavio-fernandes — CHANGES_REQUESTED** [review]
Nice! Some little nits and small fix.

File: cmd/ovsdb-mon/shell.go [hunk 4]
```diff
@@ -338,3 +373,31 @@ func colordiff(a, b interface{}) string {
...
+// filterAPI returns the conditional API that filters based on the provided filter expression
+// Expression is [FIELD]=[VALUE]
+func (s *OvsdbShell) filterAPI(tableName string, expr string) (client.ConditionalAPI, error) {
+	condModel, err := s.dbModel.NewModel(tableName)
+	if err != nil {
+		return nil, err
+	}
+	parts := strings.Split(expr, "=")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("Invalid filter expression: %s", expr)
```
[the remark this answers is not in the document]
Comment (amorenoz) [reply cmd/ovsdb-mon/shell.go:331]: Sure!

File: cmd/ovsdb-mon/shell.go (hunk 1 above)
Comment (amorenoz) [reply cmd/ovsdb-mon/shell.go:337]: LOL of course!

File: cmd/ovsdb-mon/shell.go (hunk 3 above)
Comment (amorenoz) [reply cmd/ovsdb-mon/shell.go:408]: Good catch, thanks!

File: cmd/ovsdb-mon/shell.go (hunk 2 above)
Comment (amorenoz) [reply cmd/ovsdb-mon/shell.go:236]: Done thanks!

**Review by flavio-fernandes — COMMENTED** [review]
Nice!!!
/lgtm

**Comment by lewisdenny** [comment]
@amorenoz ++ tested and working great, thank you!

**Pull request merged**